File: ground_station/js_version/commandSender.js
/**
 * commandSender.js
 * Memvalidasi dan mengirim perintah ke Raspberry Pi via serial.
 * Menggantikan command_sender.py
 */

const VALID_COMMANDS = ['ARM', 'DISARM', 'PLAN'];

class CommandSender {
  /**
   * @param {import('./serialManager')} serialManager - Instance SerialManager
   */
  constructor(serialManager) {
    this.serialManager = serialManager;
  }

  /**
   * Mengirim command ke serial port.
   * @param {string} command - Nama command: 'ARM', 'DISARM', atau 'PLAN'
   * @returns {boolean} - true jika command valid dan berhasil dikirim
   */
  send(command) {
    if (!VALID_COMMANDS.includes(command)) {
      console.warn(`[CommandSender] Command tidak valid: ${command}`);
      return false;
    }

    const msg = JSON.stringify({ type: 'command', command });
    this.serialManager.send(msg);
    console.log(`[CommandSender] Mengirim: ${msg}`);
    return true;
  }
}

module.exports = CommandSender;

File: ground_station/command_sender.py
import json

class CommandSender:
    def __init__(self, serial_manager):
        self.serial_manager = serial_manager

    def send_command(self, cmd_str: str):
        if cmd_str in ["ARM", "DISARM", "PLAN"]:
            msg = {
                "type": "command",
                "command": cmd_str
            }
            json_str = json.dumps(msg)
            self.serial_manager.send(json_str)
            return True
        return False
